### PR TITLE
first stage of refactoring

### PR DESCRIPTION
**Pull request by nikitamax**
1) removed nesting
2) cleared html
3) used map

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -1,62 +1,57 @@
...
-}
-
-// функция для обновления localStorage
 function updateLocalStorage() {
         let serialTasks = JSON.stringify(tasks);
         localStorage.setItem('tasks', serialTasks);
 }
 
-// функция для создания li с задачей
+function addChild(parent, child) {
```
[the remark this answers is not in the document]
Comment (nikitamax) [reply index.js]: function addTree(rootElement, children) {
        const fragment = document.createDocumentFragment();
        for (let i = 0; i < children.length; i++) {
                const element = document.createElement(children[i].el);
                for (let i = 0; i < children[i].attrs.length; i++) {
                        element.setAttribute(children[i].attrs[i].key, children[i].attrs[i].value);
                }
                for (let i = 0; i < children[i].children.length; i++) {
                        element.appendChild(children[i].children[i]);
                        fragment.appendChild(element);
                }
        }
        rootElement.appendChild(fragment);
}

File: index.js [hunk 2]
```diff
@@ -52,6 +53,18 @@ function addChild(parent, child) {
         parent.appendChild(child);
 }
 
+function addTree(rootElement, children) {
+        const fragment = document.createDocumentFragment();
```
Comment (golendukhin) [inline index.js]: before you run the function's logic you need to check if you actually have rootElement, do you have children and the array is not empty.

Comment (golendukhin) [reply index.js]: I would like to see attributes assignment here as well.

File: index.js [hunk 3]
```diff
@@ -252,7 +281,7 @@ function checkTask() {
                                 addButtonClear();
                         }
 
-                        if (getFilterLinks()[1].className == "selected") {
+                        if (getFilterLinks().linkActive.className == "selected") {
```
Comment (golendukhin) [inline index.js]: You should get filters only once, not each time.

File: index.js [hunk 4]
```diff
@@ -1,203 +1,109 @@
...
 
-        addTree(unList, [
-                {
-                        el: divListItem,
-                        children: [checkInput, taskName, deleteButton]
+function createFooter() {
+        let children = [
+                { el: 'footer', attrs: { className: 'footer' }, children: [
+                        {el: 'span', attrs: { className: 'todo-count' }, children: [
+                                { el: 'strong', attrs: { id: 'task-number-id' }, children: [] },
```
Comment (golendukhin) [inline index.js]: you can omit ```children: []``` and add a correspondent validation inside of addTree function

**Pull request merged**